dashboard/app/controllers/aichat_controller.rb

```diff
@@ -47,6 +47,33 @@ def chat_completion
     sagemaker_response = AichatSagemakerHelper.request_sagemaker_chat_completion(input, params[:aichatModelCustomizations][:selectedModelId])
     latest_assistant_response = AichatSagemakerHelper.get_sagemaker_assistant_response(sagemaker_response)
 
+    filter_result = ShareFiltering.find_failure(latest_assistant_response, locale)
+    if filter_result&.type == ShareFiltering::FailureType::PROFANITY
+      new_messages = [
+        {
+          role: 'user',
+          content: params[:newMessage],
+          status: SharedConstants::AI_INTERACTION_STATUS[:ERROR]
+        }
+      ]
+      session_id = log_chat_session(new_messages)
+
+      Honeybadger.notify(
```

Comment:
Is HB the right tool here? My understanding was that we want to use HoneyBadger primarily for server errors that we can triage and should try to limit/eliminate. But in this case, is this an error we would actively try to drive down?

Reply:
Yeah good q -- tl;dr is I'm not sure. Some more discussion in the Slack thread linked in [this comment](https://github.com/code-dot-org/code-dot-org/pull/58351#discussion_r1588431675) -- I'm happy to revisit the approach here down the line.